src/PerformanceTests/Common/Scenarios/LoopRunner.cs

```diff
@@ -57,7 +59,7 @@ protected override void Stop()
                     countdownEvent.Reset(BatchSize);
                     var batchDuration = Stopwatch.StartNew();
 
-                    await TaskHelper.ParallelFor(BatchSize, SendMessage);
+                    await TaskHelper.ParallelFor(BatchSize, () => SendMessage(Session));
```

Comment:
Can we change the signature of the `Loop` method to also accept the `ISession` as a parameter? That way we can completely drop the class level variable completely.